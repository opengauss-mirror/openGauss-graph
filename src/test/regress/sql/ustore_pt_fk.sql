create table pt1(a int primary key, b int)with(storage_type=ustore) partition by range(a)(partition p1 values less than(10), partition p2 values less than(100)) ;
create table pt2(a int references pt1, b int)with(storage_type=ustore) partition by range(a)(partition p1 values less than(10), partition p2 values less than(100)) ;

insert into pt2 values(1,1);
insert into pt1 values(1,1);
insert into pt2 values(1,1);
insert into pt2 values(2,2);

update pt2 set a=2,b=2;

insert into pt1 values(2,2);
update pt2 set a=2,b=2;


insert into pt2 values(3,3);
update pt1 set a=3,b=3 where a=1;

insert into pt2 values(3,3);

delete from pt1;

insert into pt1 values(4,4);
delete from pt1 where a=4;


insert into pt1 values(20,20);
insert into pt2 values(20,20);

delete from pt2;
delete from pt1;

drop table pt2 cascade;
drop table pt1 cascade;