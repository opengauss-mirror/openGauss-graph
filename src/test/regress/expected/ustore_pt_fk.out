create table pt1(a int primary key, b int)with(storage_type=ustore) partition by range(a)(partition p1 values less than(10), partition p2 values less than(100)) ;
NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "pt1_pkey" for table "pt1"
create table pt2(a int references pt1, b int)with(storage_type=ustore) partition by range(a)(partition p1 values less than(10), partition p2 values less than(100)) ;
insert into pt2 values(1,1);
ERROR:  insert or update on table "pt2" violates foreign key constraint "pt2_a_fkey"
DETAIL:  Key (a)=(1) is not present in table "pt1".
insert into pt1 values(1,1);
insert into pt2 values(1,1);
insert into pt2 values(2,2);
ERROR:  insert or update on table "pt2" violates foreign key constraint "pt2_a_fkey"
DETAIL:  Key (a)=(2) is not present in table "pt1".
update pt2 set a=2,b=2;
ERROR:  insert or update on table "pt2" violates foreign key constraint "pt2_a_fkey"
DETAIL:  Key (a)=(2) is not present in table "pt1".
insert into pt1 values(2,2);
update pt2 set a=2,b=2;
insert into pt2 values(3,3);
ERROR:  insert or update on table "pt2" violates foreign key constraint "pt2_a_fkey"
DETAIL:  Key (a)=(3) is not present in table "pt1".
update pt1 set a=3,b=3 where a=1;
insert into pt2 values(3,3);
delete from pt1;
ERROR:  update or delete on table "pt1" violates foreign key constraint "pt2_a_fkey" on table "pt2"
DETAIL:  Key (a)=(2) is still referenced from table "pt2".
insert into pt1 values(4,4);
delete from pt1 where a=4;
insert into pt1 values(20,20);
insert into pt2 values(20,20);
delete from pt2;
delete from pt1;
drop table pt2 cascade;
drop table pt1 cascade;
